helix-view/src/editor/variable_expansion.rs

```diff
@@ -0,0 +1,149 @@
+use std::borrow::Cow;
+
+use crate::Editor;
+
+pub const VARIABLES: [&str; 7] = [
+    "%sh{}",
+    "%{basename}",
+    "%{filename}",
+    "%{dirname}",
+    "%{cwd}",
+    "%{linenumber}",
+    "%{selection}",
+];
+
+pub fn expand_variables<'a>(editor: &Editor, input: &'a str) -> anyhow::Result<Cow<'a, str>> {
+    let (view, doc) = current_ref!(editor);
+    let shell = &editor.config().shell;
+
+    let mut output: Option<String> = None;
+
+    let mut chars = input.char_indices();
+    let mut last_push_end: usize = 0;
+
+    while let Some((index, char)) = chars.next() {
+        if char == '%' {
+            if let Some((_, char)) = chars.next() {
+                if char == '{' {
+                    for (end, char) in chars.by_ref() {
+                        if char == '}' {
+                            if output.is_none() {
+                                output = Some(String::with_capacity(input.len()))
+                            }
+
+                            if let Some(o) = output.as_mut() {
+                                o.push_str(&input[last_push_end..index]);
+                                last_push_end = end + 1;
+
+                                let value = match &input[index + 2..end] {
+                                    "basename" => doc
+                                        .path()
+                                        .and_then(|it| it.file_name().and_then(|it| it.to_str()))
+                                        .unwrap_or(crate::document::SCRATCH_BUFFER_NAME)
+                                        .to_owned(),
+                                    "filename" => doc
+                                        .path()
+                                        .and_then(|it| it.to_str())
+                                        .unwrap_or(crate::document::SCRATCH_BUFFER_NAME)
+                                        .to_owned(),
+                                    "dirname" => doc
+                                        .path()
+                                        .and_then(|p| p.parent())
+                                        .and_then(std::path::Path::to_str)
+                                        .unwrap_or(crate::document::SCRATCH_BUFFER_NAME)
+                                        .to_owned(),
+                                    "cwd" => std::env::current_dir()?.to_str().unwrap().to_owned(),
```

Comment:
```suggestion
                                    "cwd" => helix_std::env::current_working_dir().to_str().unwrap().to_owned(),
```

that function will be available if you rebase/merge-master

Reply:
(you probably meant `helix_stdx::env::....`, not `helix_std::env::....`)